src/enforester.js

```diff
@@ -1314,17 +1327,26 @@ export class Enforester {
   enforestNewExpression() {
     this.matchKeyword('new');
     let callee;
-    if (this.isKeyword(this.peek(), 'new')) {
+    let lookahead = this.peek();
+    if (this.isKeyword(lookahead, 'new')) {
       callee = this.enforestNewExpression();
-    } else if (this.isKeyword(this.peek(), 'super')) {
+    } else if (this.isKeyword(lookahead, 'super')) {
+      callee = this.enforestLeftHandSideExpression({ allowCall: false });
+    } else if (this.isParens(lookahead) || this.isBrackets(lookahead) || this.isBraces(lookahead) || this.isTemplate(lookahead)) {
       callee = this.enforestExpressionLoop();
```

Comment:
I can move the logic in the above four lines to `enforestLeftHandSideExpression`. Everything else handles `new` specific edge cases.